src/node/api/db.ts

```diff
@@ -0,0 +1,166 @@
+import { Hex, keccak256 } from 'viem';
+import { z } from 'zod';
+import { Storage } from '../../storage/index.js';
+
+/**
+ * Interface defining the properties of a User object stored in storage.
+ */
+export interface User {
+  /** The username of the user */
+  login: string;
+  /** The hashed password of the user */
+  hashedPassword: string;
+  /** Flag which indicate that the use is admin */
+  isAdmin: boolean;
+  /** The optional JSON Web Token of the user */
+  jwt?: string;
+}
+
+/**
+ * Validation schema for user creation API input.
+ * It validates that the input contains a nonempty `login` and `password`.
+ */
+export const UserInputSchema = z.object({
+  login: z.string().nonempty(),
+  password: z.union([z.string().nonempty(), z.string().startsWith('0x')]),
+});
+
+/**
+ * Type definition for User registration input,
+ * inferred from UserInputSchema.
+ */
+export type UserInputType = z.infer<typeof UserInputSchema>;
+
+/**
+ * Interface defining the properties of UsersDb initialization options.
+ */
+export interface UsersDbOptions {
+  /** Instance of storage used for persisting the state of the API server */
+  storage: Storage;
+  /** Prefix used for the storage key to avoid potential key collisions */
+  prefix: string;
+  /** Salt used for hashing passwords */
+  salt: string;
+}
+
+export class UsersDb {
+  /** Storage instance for persisting the state of the API server */
+  storage: Storage;
+  /** Specific key prefix for the storage key to avoid potential key collisions */
+  prefix: string;
+  /** Salt used for hashing passwords */
+  salt: string;
+
+  /**
+   * Creates an instance of UsersDb.
+   * Initializes an instance of UsersDb with given options.
+   *
+   * @param {UsersDbOptions} options
+   * @memberof NodeApiServer
+   */
+  constructor(options: UsersDbOptions) {
+    const { storage, prefix, salt } = options;
+
+    // TODO Validate NodeApiServerOptions
+
+    this.prefix = `${prefix}_api_users_`;
+    this.storage = storage;
+    this.salt = salt;
+  }
+
+  /**
+   * Hashes the given password with the given salt.
+   *
+   * @static
+   * @param {string} password The password to be hashed
+   * @param {string} salt The salt to be used for hashing
+   * @returns {string} The hashed password
+   * @memberof UsersDb
+   */
+  static hashPassword(password: string, salt: string): string {
+    return keccak256(`${password}${salt}` as Hex);
```

Comment:
```
const rounds = 2;
const hashedPassword = await bcrypt.hash(String(password), rounds);

await bcrypt.compare(password, user.password)
```
I think bcrypt is more secure for passwords 
https://www.npmjs.com/package/bcrypt
(We can use abstract rounds or store salt for every user and use hashing with salt)